File: VB/WebApplication1/XtraReport1.vb
Imports DevExpress.XtraReports.UI

Namespace WebApplication1
    Partial Public Class XtraReport1
        Inherits DevExpress.XtraReports.UI.XtraReport

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub lbShowHide_BeforePrint(ByVal sender As Object, ByVal e As System.Drawing.Printing.PrintEventArgs) Handles lbShowHide.BeforePrint
            TryCast(sender, XRLabel).Text = If(DetailReport.DrillDownExpanded, "Hide Details", "Show Details")
        End Sub

    End Class
End Namespace

File: CS/WebApplication1/XtraReport1.cs
using DevExpress.XtraReports.UI;

namespace WebApplication1 {
    public partial class XtraReport1 : DevExpress.XtraReports.UI.XtraReport {
        public XtraReport1() {
            InitializeComponent();
        }

        private void lbShowHide_BeforePrint(object sender, System.Drawing.Printing.PrintEventArgs e) {
            (sender as XRLabel).Text = DetailReport.DrillDownExpanded ? "Hide Details" : "Show Details";
        }

    }
}
